packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  withExperimentalFeatures,
+  createNewStory,
+  visitAdminPage,
+  insertStoryTitle,
+  publishStory,
+  takeSnapshot,
+  addCustomFont,
+  removeCustomFont,
+} from '@web-stories-wp/e2e-test-utils';
+
+const createStoryWithTitle = async (title) => {
+  await createNewStory();
+  await insertStoryTitle(title);
+  await expect(page).toClick('#library-tab-text');
+  const insertButton = await page.waitForXPath(
+    `//button//span[contains(text(), 'Title 1')]`
+  );
+  await insertButton.click();
+  await expect(page).toMatchElement('[data-testid="textFrame"]', {
+    text: 'Title 1',
+  });
+};
+
+const updateFont = async (fontFamily) => {
+  await expect(page).toClick('button[aria-label="Font family"]');
+  await page.keyboard.type(fontFamily);
+  await page.keyboard.press('ArrowDown');
+  await page.keyboard.press('Enter');
+};
+
+const visitStoryWithTitle = async (title) => {
+  await visitAdminPage('edit.php', 'post_type=web-story');
+  await expect(page).toClick('a', { text: title });
+};
+
+const replaceFontWithFontPicker = async (fontFamily = '') => {
+  await page.waitForSelector('[role="dialog"]');
+  await expect(page).toMatchElement('button', { text: 'Open anyway' });
+  await expect(page).toMatchElement('button', { text: 'Replace font' });
+  takeSnapshot('Missing fonts dialog');
+  await expect(page).toClick(
+    "div[role='dialog'] div[class^='datalist__'] button"
+  );
+  await page.keyboard.type(fontFamily);
+  await page.keyboard.press('ArrowDown');
+  takeSnapshot('Missing fonts dialog font picker');
+  await page.keyboard.press('Enter');
+  await expect(page).toClick('button', { text: 'Replace font' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+const replaceFontUsingDefault = async () => {
+  await page.waitForSelector('[role="dialog"]');
+  await expect(page).toClick('button', { text: 'Open anyway' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+describe('Font Check', () => {
+  describe('Font Check Dialog', () => {
+    withExperimentalFeatures(['customFonts']); // 'notifyDeletedFonts'
+
+    it('should show dialog & replace it with selected font', async () => {
```

Comment:
should show dialog & replace it with selected font and should show dialog & replace font with default font seem like very similar tests. By are both needed and what is the difference?

Reply:
One is for replacing the font with one from the font picker, the other is for replacing with our default font, Roboto (that happens when you click on "Open anyway").

Reply:
I more meant from a code standard, what is the difference. There is lots of repeated code here. Can we make some of this into a function with params?

Reply:
I'll make some adjustments to cut down on some of this.